dashboard/test/models/pd/enrollment_test.rb

```diff
@@ -33,7 +33,7 @@ class Pd::EnrollmentTest < ActiveSupport::TestCase
     enrollment_with_user = create :pd_enrollment, user: teacher2
     enrollment_with_no_user = create :pd_enrollment
 
-    assert_nil enrollment_with_email.user
+    assert_equal teacher1, enrollment_with_email.user
```

Comment:
This now doesn't really test resolve_user. I'd prefer to changes these `creates` to `builds` so the new method isn't run here. Then add a new test case for the before_validation hook, which explicitly calls `valid?`